src/makeCalculator.js

```diff
@@ -4,7 +4,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+  };
+
+  calculator.add = function (value) {
+    return this.result + value;
+  }.bind(calculator);
+
+  calculator.subtract = function (value) {
+    return this.result - value;
+  }.bind(calculator);
+
+  calculator.multiply = function (value) {
+    return this.result * value;
+  }.bind(calculator);
+
+  calculator.divide = function (value) {
+    return this.result / value;
+  }.bind(calculator);
+
+  calculator.reset = function () {
+    this.result = 0;
+
+    return this;
+  }.bind(calculator);
+
+  calculator.operate = function (callback, value) {
+    this.result = callback(value);
+
+    return this;
+  }.bind(calculator);
```

Comment:
[CODE KNOWLEDGE] - Use the method shorthand syntax for the `operate` method. Also, ensure that the callback is a function before attempting to call it.

GOOD EXAMPLE:
operate(callback, value) {
  if (typeof callback === 'function') {
    this.result = callback(this.result, value);
  }
  return this;
},